test/integration/sanity/vaultSecrets.test.js

```diff
@@ -992,5 +992,165 @@ describe('vaultSecrets', function () {
                 expect(prerequest.vaultSecrets).to.deep.equal(undefined);
             });
         });
+
+        describe('should handle _allowScriptAccess as a function', function () {
```

Comment:
We should refactor all exisiting tests to have `_allowScriptAccess` as function and remove these new ones as it will be redundant. 

Keep a test with `_allowScriptAccess` being not function and one more where it throws.